users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -106,13 +117,93 @@ def add_specaug_layer(
     # Update config with pre-generated parameters
     full_config["specaug_params"] = specaug_params
 
-    network[name] = {
-        "class": "eval",
-        "from": from_list,
-        "eval": f"self.network.get_config().typed_value('transform')(source(0, as_data=True), network=self.network, **{full_config})",
-    }
+    if not full_config["mask_based_on_variance"]:
+        network[name] = {
+            "class": "eval",
+            "from": from_list,
+            "eval": f"self.network.get_config().typed_value('transform')(source(0, as_data=True), network=self.network, **{full_config})",
+        }
+        return [name], [
+            sort_filters_by_center_freq,
+            get_frequency_response,
+            _mask,
+            random_mask,
+        ]
+    else:
+        network[name] = {
+            "class": "eval",
+            "from": from_list,
+            "eval": f"self.network.get_config().typed_value('transform_with_variance')(source(0, as_data=True), network=self.network, **{full_config})",
+        }
+        return [name], [
+            get_frequency_response,
+            calculate_frequency_response_variance,
+            _mask,
+            random_mask,
+            filter_based_masking ,
+            transform_with_variance,
+        ]
+
+
+def calculate_frequency_response_variance(filter_layer):
```

Comment:
It isn't necessary to introduce this function imho. Instead of calling this function, you just have two lines that are very easy to read:
```
    f_resp = get_frequency_response(filter_layer)
    f_resp_var = tf.math.reduce_variance(f_resp, axis=0)
```